firmware/examples/PWM/source/main.cpp

```diff
@@ -0,0 +1,37 @@
+#include <cstdarg>
+#include <cstdio>
+#include <cstdint>
+#include "L0_LowLevel/delay.hpp"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/pwm.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+int main(void)
+{
+    DEBUG_PRINT("Pwm Application Starting...\n");
+
+    // Standard gpio powered LED
+    DEBUG_PRINT("Creating Gpio powered LED...\n");
```

Comment:
This should explain why you are creating GPIO powered LED. When I first read this, I had no idea why this was included in this code. Remember that the audience reading this are CmpE 30 students (new at programming) and people unfamiliar with the framework. You need to actually spell out what it is that you are doing at each step, so that the user fully understands every portion of the code. 

See PinConfigure example as a template.

Reply:
You should also guide the user, to how they should setup the pin with an led and resistor. You could tell them to use a 470 ohm resistor for both gpio and pwm led. Then tell them that the GPIO represents the maximum brightness that the led can be.